### ISSUE TITLE
alsa bug (crash)

### ISSUE THREAD

**Issue by user706** [pure-data/pure-data#969]
Issuing the command 
`pd -alsa -alsaadd default -audioadddev default -alsamidi -noadc -midioutdev 1`
and then starting Media->Test audio and MIDI
very often results in 
```
restart alsa output
alsa xrun recovery apparently failed
pd: pcm.c:1168: snd_pcm_prepare: Assertion `pcm' failed.
Pd: signal 6
```

The reason for this is the following bug:
https://github.com/pure-data/pure-data/commit/75819aad31bedc8466858fb81fe69c97522a82ed#r38730555

**Comment by millerpuckette** [comment]
Thanks, should be fixed now

**Comment by umlaeute** [comment]
fixed with 80929213b540af1ba12c2c8abd7a58d0c1eb2204
